client seg fault: invalid memory address or nil pointer dereference

Have seen this a few times while running strest-grpc in a k8s test environment. The strest client is being run with:

```
/go/bin/strest-client \
  -address $NODE_NAME:5757 \
  -concurrency 500 \
  -interval 30s \
  -latencyPercentiles 50=0,100=0 \
  -lengthPercentiles 50=0,100=0 \
```

And it occasionally exits with:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x7e22f9]

goroutine 197 [running]:
main.sendNonStreamingRequests(0xaf9940, 0xc4205c82b8, 0xc4202d6720, 0xc42016c1e0, 0xc42016c180, 0xc420fadf60, 0xc420172360, 0x0, 0x0)
	/go/src/github.com/buoyantio/strest-grpc/client/main.go:181 +0x209
main.main.func3(0xc42015c070, 0xc42015c0ad, 0xc4202d6720, 0xc42016c1e0, 0xc42016c180, 0xc420172360, 0xc42015c0d0, 0xc42015c1c0)
	/go/src/github.com/buoyantio/strest-grpc/client/main.go:381 +0x381
created by main.main
	/go/src/github.com/buoyantio/strest-grpc/client/main.go:393 +0xb31
```

--- a/client/main.go
+++ b/client/main.go
@@ -168,26 +168,26 @@ func sendNonStreamingRequests(client pb.ResponderClient,
 	shutdownChannel <-chan struct{},
 	lengthDistribution distribution.Distribution,
 	latencyDistribution distribution.Distribution, r *rand.Rand,
-	received chan *MeasuredResponse) error {
+	received chan *MeasuredResponse) {
 	for {
 		select {
 		case <-shutdownChannel:
-			return nil
+			return
 		default:
-
 			start := time.Now()
 			resp, err := client.Get(context.Background(),
 				&pb.ResponseSpec{
 					Length:  int32(lengthDistribution.Get(r.Int31() % 1000)),
 					Latency: latencyDistribution.Get(r.Int31() % 1000)})
 			if err != nil {
-				return err
+				received <- &MeasuredResponse{err: err}
+				continue
 			}
 
 			bytes := int64(len([]byte(resp.Body)))
 			latency := time.Since(start)
 			promLatencyHistogram.Observe(float64(latency))
-			received <- &MeasuredResponse{0, latency, bytes, err}
+			received <- &MeasuredResponse{latency: latency, bytes: bytes}
 		}
 	}
 }
@@ -377,11 +377,8 @@ func main() {
 			client := pb.NewResponderClient(conn)
 
 			if !*streaming {
-				err := sendNonStreamingRequests(client, shutdownChannel,
+				sendNonStreamingRequests(client, shutdownChannel,
 					lengthDistribution, latencyDistribution, r, received)
-				if err != nil {
-					log.Fatalf("could not send a request: %v", err)
-				}
 			} else {
 				err := sendStreamingRequests(client, shutdownChannel,
 					lengthDistribution, latencyDistribution, *streamingRatio, r, received)

